tests/test_restriction_matrix.cc

```diff
@@ -334,6 +340,6 @@ BOOST_AUTO_TEST_CASE(weight_sum, *utf::tolerance(1e-4))
       e[i] = 1.;
     e.compress(::dealii::VectorOperation::insert);
     restrictor_matrix.vmult(ee, e);
-    BOOST_TEST(ee.l1_norm() == 1.);
+    BOOST_TEST(ee.l1_norm() == 1., tt::tolerance(1e-4));
```

Comment:
What is the rational for having the floating point comparison manipulation in the `BOOST_TEST` macro versus having a decorator for the unit test?

Reply:
I was not sure that `BOOST_AUTO_TEST_CASE` and `BOOST_DATA_TEST_CASE` behave the same way, and was too lazy to google. So just replaced with what I know.